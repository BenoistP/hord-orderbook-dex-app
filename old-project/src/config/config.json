{
  "network": 97,
  "rpcUrl": "https://data-seed-prebsc-1-s1.binance.org:8545/",
  "thegraphUrl": "https://api.thegraph.com/subgraphs/name/hord/hord-develop",
  "apiPath": "https://test-api.hord.app",
  "stakeAmount": "1000",
  "daysToStake": "10",
  "stage": "develop"
}
