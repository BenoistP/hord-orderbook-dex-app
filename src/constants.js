export const LIVE_NET_ID = '1';
export const KOVAN_NET_ID = '42';
