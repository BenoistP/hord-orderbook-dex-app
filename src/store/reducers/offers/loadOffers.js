import {createPromiseActions} from '../../../utils/createPromiseActions';
import {syncOffer} from './syncOfferEpic';
import {getBestOffer, getWorseOffer} from './syncOffersEpic';

export const loadSellOffers = createPromiseActions('OFFERS/LOAD_SELL_OFFERS');
export const loadSellOffersEpic = (offerCount, sellToken, buyToken, {
  doGetBestOffer = getBestOffer,
  doSyncOffer = syncOffer,
  doGetWorseOffer = getWorseOffer,
} = {}) => async (dispatch) => {
  const sellOffersTradingPair = { baseToken: sellToken, quoteToken: buyToken };
  try {
    dispatch(loadSellOffers.pending(sellOffersTradingPair));
    let currentSellOfferId = (await dispatch(doGetBestOffer(sellToken, buyToken))).value.toNumber();
    while (currentSellOfferId !== 0) {
      dispatch(doSyncOffer(currentSellOfferId));
      currentSellOfferId = (await dispatch(doGetWorseOffer(currentSellOfferId))).value.toNumber()
    }
    dispatch(loadSellOffers.fulfilled(sellOffersTradingPair));
    return loadSellOffers;
  } catch(e) {
    dispatch(loadSellOffers.rejected(e));
  }
};

export const loadBuyOffers = createPromiseActions('OFFERS/LOAD_BUY_OFFERS');
export const loadBuyOffersEpic = (offerCount, sellToken, buyToken, {
  doGetBestOffer = getBestOffer,
  doSyncOffer = syncOffer,
  doGetWorseOffer = getWorseOffer,
} = {}) => async (dispatch) => {
  const buyOffersTradingPair = { baseToken: sellToken, quoteToken: buyToken };
  try {
    dispatch(loadBuyOffers.pending(buyOffersTradingPair));
    let currentBuyOfferId = (await dispatch(doGetBestOffer(buyToken, sellToken))).value.toNumber();
    while (currentBuyOfferId !== 0) {
      dispatch(doSyncOffer(currentBuyOfferId));
      currentBuyOfferId = (await dispatch(doGetWorseOffer(currentBuyOfferId))).value.toNumber();
    }
    dispatch(loadBuyOffers.fulfilled(buyOffersTradingPair));
    return loadBuyOffers;
  } catch(e) {
    dispatch(loadBuyOffers.rejected(e));
  }
};
