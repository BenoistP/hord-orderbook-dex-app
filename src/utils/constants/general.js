export const LS_ACCOUNT = '555cg03294173ace0a7d925d8rc9921513e333ad'
export const CH_ACCOUNT = '285cg03294173ace907d925d8rc99215139299kl'

export const MESSAGE_BOX_TYPES = {
  ERROR: 'error',
  SUCCESS: 'success',
}

export const EMPTY_BYTE_ARRAY = '0x00000000000000000000000000000000'
export const EMPTY_HEX = '0000000000000000000000000000000000000000000000000000000000000000'

export const HORD_SYMBOL = 'HORD'

export const twitterLink = 'https://twitter.com/HordApp'
export const telegramLink = 'https://t.me/hordnews'
export const mediumLink = 'https://medium.com/hord-app'
export const hordAppLink = 'https://www.hord.app/'
export const docsLink = 'https://docs.hord.app/'

export const FOOTER_LINKS = [
  { name: 'Hord.app', link: hordAppLink },
  { name: 'Telegram', link: telegramLink },
  { name: 'Twitter', link: twitterLink },
  { name: 'Medium', link: mediumLink },
  { name: 'Docs', link: docsLink },
]

export const FOOTER_TERMS_POLICY = [
  { name: 'common.terms', link: '' },
  { name: 'common.policy', link: '' },
]
