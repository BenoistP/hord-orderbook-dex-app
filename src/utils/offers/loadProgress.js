import React from "react";
import PropTypes from "prop-types";
import ImmutablePropTypes from "react-immutable-proptypes";
import OasisIcon from "../../components/OasisIcon";

const loadProgress = (loadedOffersList, offersTotalCount) => {
  const progress = loadedOffersList && offersTotalCount
    ? parseInt(loadedOffersList.count() / offersTotalCount * 100)
    : null;
  return progress <= 100 ?  progress: 100;
};

const style = (loadedOffersList, offersTotalCount, showPercent) => {
  const progress = showPercent ? loadProgress(loadedOffersList, offersTotalCount) : 100;
  return {
    background: `linear-gradient(90deg, #F8F7F5 ${
      progress ? progress : 0
    }%, #fff 0%)`,
    border: "1px solid #b1b1b3",
    color: "#68686C",
    fontSize: "11px",
    padding: "5px 0",
    float: "right",
    fontWeight: "bold",
    borderRadius: "7%",
    width: "130px",
    textAlign: "center",
    letterSpacing: "normal",
    position: "relative",
    top: '1px'
  };
};

const LoadProgressSection = ({ loadedOffersList, offersTotalCount, showPercent }) => {
  const loaded = loadProgress(loadedOffersList, offersTotalCount);
  return offersTotalCount == null ? (
    <span style={{ float: "right" }}>
      <OasisIcon icon="loading" />
    </span>
  ) : (
    <span
      style={style(loadedOffersList, offersTotalCount)}
      className={"loadProgress"}
    >
      <span hidden={offersTotalCount != null}>
        <OasisIcon icon="loading" />
      </span>
      <span hidden={loaded == null}>
        {showPercent ? `${loaded}% / `: null}  {offersTotalCount}
      </span>
      <span hidden={loaded != null && offersTotalCount != null}>
        {offersTotalCount === 0 ? 'no orders' : offersTotalCount }
      </span>
    </span>
  );
};

LoadProgressSection.propTypes = {
  loadedOffersList: ImmutablePropTypes.list,
  offersTotalCount: PropTypes.number,
  showPercent: PropTypes.bool
};
export { loadProgress, LoadProgressSection };
