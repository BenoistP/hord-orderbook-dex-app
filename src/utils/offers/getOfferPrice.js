import { MAKE_BUY_OFFER, MAKE_SELL_OFFER } from "../../constants";
import web3 from "../../bootstrap/web3";

export const getOfferPrice = (
  offer,
  offerMakeType,
  formats = { asString: true, asSortValue: true, asBigNumber: false }
) => {
  switch (offerMakeType) {
    case MAKE_BUY_OFFER:
      return {
        asString: offer.bid_price,
        asSortValue: offer.bid_price_sort,
        asBN: formats.asBigNumber ? web3.toBigNumber(offer.bid_price): null
      };
    case MAKE_SELL_OFFER:
      return {
        asString: offer.ask_price,
        asSortValue: offer.ask_price_sort,
        asBN: formats.asBigNumber ? web3.toBigNumber(offer.bid_price) : null
      };
  }
};
