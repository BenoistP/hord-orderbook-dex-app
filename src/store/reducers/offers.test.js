/* global describe test expect jest */

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import each from 'jest-each';

import BigNumber from 'bignumber.js';
import {Map, List} from 'immutable';

import config from '../../configs';
import offers from './offers';
import {OFFER_SYNC_TYPE_INITIAL, OFFER_SYNC_TYPE_NEW_OFFER, OFFER_SYNC_TYPE_UPDATE} from './offers';
import {TYPE_BUY_OFFER} from "./offers";
import conversion from '../../utils/conversion';
import {SYNC_STATUS_PRISTINE} from '../../constants';

const TOKEN_ADDRS = config.tokens.kovan;

each([
  ['initial sync', OFFER_SYNC_TYPE_INITIAL],
  ['new offer sync', OFFER_SYNC_TYPE_NEW_OFFER],
  ['update sync', OFFER_SYNC_TYPE_UPDATE],
]).describe('syncOffer', (description, syncType) => {

  test(description, async () => {
    const store = configureMockStore([thunk])(Map({
      tokens: Map({
        precision: 18,
        tokenSpecs: Map({
          "MKR": Map({}),
          "W-ETH": Map({}),
        }),
        tradingPairs: List([
          Map({
            base: "MKR",
            quote: "W-ETH",
          }),
        ]),
      }),
    }));

    const setOfferEpic = jest.fn(() => () => null);
    const getTradingPairOfferCount = jest.fn(() => () => null);
    const promise = store.dispatch(offers.testActions.syncOffer('61209', syncType, null, {
      doLoadOffer: () => async () => ({value: [new BigNumber("130350000000000000"), TOKEN_ADDRS["W-ETH"], new BigNumber("100000000000000000"), TOKEN_ADDRS["MKR"], "0x2495eb6895c2e6c591ae9eb63a07b4a450623220", false, new BigNumber("0")]}),
      doGetTradingPairOfferCount: getTradingPairOfferCount,
      doSetOfferEpic: setOfferEpic,
    }));

    const result = await promise;

    expect(result).toMatchSnapshot();
    expect(store.getActions()).toMatchSnapshot();
    expect(getTradingPairOfferCount.mock.calls).toMatchSnapshot();
    expect(setOfferEpic.mock.calls).toMatchSnapshot();
  });

});

each([
  ['new offer sync / set', OFFER_SYNC_TYPE_NEW_OFFER],
  ['initial sync / update', OFFER_SYNC_TYPE_INITIAL],
  ['initial sync / set', OFFER_SYNC_TYPE_INITIAL, {id: 61210}],
  ['update sync / partial', OFFER_SYNC_TYPE_UPDATE],
  ['update sync / complete', OFFER_SYNC_TYPE_UPDATE, {buyHowMuch: new BigNumber(0)}],
]).describe('setOfferEpic', (description, syncType, overrides = {}) => {

  test(description, async () => {
    const store = configureMockStore([thunk])(Map({
      tokens: Map({
        precision: 18,
        tokenSpecs: Map({
          "MKR": Map({
            precision: 18,
          }),
          "W-ETH": Map({
            precision: 18,
          }),
        }),
      }),
      offers: Map({
        offers: Map()
          .set(Map({baseToken: 'MKR', quoteToken: 'W-ETH'}), Map({
            buyOfferCount: 1,
            buyOffers: List([
              {id: 61209},
            ]),
          })),
      }),
    }));
    conversion.init(store.getState);

    const promise = store.dispatch(offers.testActions.setOfferEpic(Object.assign({
      id: 61209,
      sellHowMuch: new BigNumber("130350000000000000"),
      sellWhichTokenAddress: TOKEN_ADDRS["W-ETH"],
      buyHowMuch: new BigNumber("100000000000000000"),
      buyWhichTokenAddress: TOKEN_ADDRS["MKR"],
      owner: "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
      status: undefined,
      offerType: TYPE_BUY_OFFER,
      syncType: syncType,
      tradingPair: {baseToken: "MKR", quoteToken: "W-ETH"},
      previousOfferState: undefined,
    }, overrides)));

    const result = await promise;

    expect(result).toMatchSnapshot();
    expect(store.getActions()).toMatchSnapshot();
  });

});

each([
  ['buy(0)', offers.testActions.loadBuyOffersEpic, []],
  ['buy(1)', offers.testActions.loadBuyOffersEpic, [900]],
  ['buy(2)', offers.testActions.loadBuyOffersEpic, [800, 700]],
  ['sell(0)', offers.testActions.loadSellOffersEpic, []],
  ['sell(1)', offers.testActions.loadSellOffersEpic, [900]],
  ['sell(2)', offers.testActions.loadSellOffersEpic, [800, 700]],
]).describe('load(Buy/Sell)OffersEpic', (description, action, offerIds) => {

  test(description, async () => {
    const store = configureMockStore([thunk])({});

    const getOfferIds = offerIds.reduce((a, id) => a.mockReturnValueOnce(id), jest.fn()).mockReturnValue(0);
    const syncOffer = jest.fn(() => () => null);
    const promise = store.dispatch(action({buyOfferCount: 100, sellOfferCount: 100}, "MKR", "W-ETH", {
      doGetBestOffer: () => async () => ({value: new BigNumber(getOfferIds())}),
      doSyncOffer: syncOffer,
      doGetWorseOffer: () => async () => ({value: new BigNumber(getOfferIds())}),
    }));

    const result = await promise;

    expect(result).toMatchSnapshot();
    expect(store.getActions()).toMatchSnapshot();
    expect(syncOffer.mock.calls).toMatchSnapshot();
  });

});

describe('syncOffersEpic', () => {

  test('main', async () => {
    const store = configureMockStore([thunk])(Map({
      tokens: Map({
        activeTradingPair: Map({
          baseToken: "MKR",
          quoteToken: "W-ETH",
        }),
      }),
      offers: Map({
        offers: Map()
          .set(Map({baseToken: 'MKR', quoteToken: 'W-ETH'}), Map({
            initialSyncStatus: SYNC_STATUS_PRISTINE,
          })),
      }),
    }));

    const loadBuyOffersEpic = jest.fn(() => async () => null);
    const loadSellOffersEpic = jest.fn(() => async () => null);
    const promise = store.dispatch(offers.actions.syncOffersEpic({baseToken: "MKR", quoteToken: "W-ETH"}, {
      doGetTradingPairOfferCount: () => async () => ({value: 10}),
      doLoadBuyOffersEpic: loadBuyOffersEpic,
      doLoadSellOffersEpic: loadSellOffersEpic,
    }));

    const result = await promise;

    expect(result).toMatchSnapshot();
    expect(store.getActions()).toMatchSnapshot();
    expect(loadBuyOffersEpic.mock.calls).toMatchSnapshot();
    expect(loadSellOffersEpic.mock.calls).toMatchSnapshot();
  });

});
