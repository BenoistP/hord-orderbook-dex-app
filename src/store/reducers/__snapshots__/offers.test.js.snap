// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load(Buy/Sell)OffersEpic buy(0) 1`] = `
Object {
  "fulfilled": [Function],
  "pending": [Function],
  "rejected": [Function],
}
`;

exports[`load(Buy/Sell)OffersEpic buy(0) 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/LOAD_BUY_OFFERS_PENDING",
  },
]
`;

exports[`load(Buy/Sell)OffersEpic buy(0) 3`] = `Array []`;

exports[`load(Buy/Sell)OffersEpic buy(1) 1`] = `
Object {
  "fulfilled": [Function],
  "pending": [Function],
  "rejected": [Function],
}
`;

exports[`load(Buy/Sell)OffersEpic buy(1) 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/LOAD_BUY_OFFERS_PENDING",
  },
]
`;

exports[`load(Buy/Sell)OffersEpic buy(1) 3`] = `
Array [
  Array [
    900,
  ],
]
`;

exports[`load(Buy/Sell)OffersEpic buy(2) 1`] = `
Object {
  "fulfilled": [Function],
  "pending": [Function],
  "rejected": [Function],
}
`;

exports[`load(Buy/Sell)OffersEpic buy(2) 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/LOAD_BUY_OFFERS_PENDING",
  },
]
`;

exports[`load(Buy/Sell)OffersEpic buy(2) 3`] = `
Array [
  Array [
    800,
  ],
  Array [
    700,
  ],
]
`;

exports[`load(Buy/Sell)OffersEpic sell(0) 1`] = `
Object {
  "fulfilled": [Function],
  "pending": [Function],
  "rejected": [Function],
}
`;

exports[`load(Buy/Sell)OffersEpic sell(0) 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/LOAD_SELL_OFFERS_PENDING",
  },
]
`;

exports[`load(Buy/Sell)OffersEpic sell(0) 3`] = `Array []`;

exports[`load(Buy/Sell)OffersEpic sell(1) 1`] = `
Object {
  "fulfilled": [Function],
  "pending": [Function],
  "rejected": [Function],
}
`;

exports[`load(Buy/Sell)OffersEpic sell(1) 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/LOAD_SELL_OFFERS_PENDING",
  },
]
`;

exports[`load(Buy/Sell)OffersEpic sell(1) 3`] = `
Array [
  Array [
    900,
  ],
]
`;

exports[`load(Buy/Sell)OffersEpic sell(2) 1`] = `
Object {
  "fulfilled": [Function],
  "pending": [Function],
  "rejected": [Function],
}
`;

exports[`load(Buy/Sell)OffersEpic sell(2) 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/LOAD_SELL_OFFERS_PENDING",
  },
]
`;

exports[`load(Buy/Sell)OffersEpic sell(2) 3`] = `
Array [
  Array [
    800,
  ],
  Array [
    700,
  ],
]
`;

exports[`setOfferEpic initial sync / set 1`] = `undefined`;

exports[`setOfferEpic initial sync / set 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "offer": Object {
        "ask_price": "0.76716532412734944381",
        "ask_price_sort": 0.7671653241273494,
        "bid_price": "1.3035",
        "bid_price_sort": 1.3035,
        "buyHowMuch": "100000000000000000",
        "buyHowMuch_filter": 100000000000000000,
        "buyWhichToken": "MKR",
        "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
        "id": 61210,
        "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
        "sellHowMuch": "130350000000000000",
        "sellHowMuch_filter": 130350000000000000,
        "sellWhichToken": "W-ETH",
        "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
        "status": undefined,
      },
      "offerType": "OFFERS/TYPE_BUY",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/SET_OFFER",
  },
]
`;

exports[`setOfferEpic initial sync / update 1`] = `undefined`;

exports[`setOfferEpic initial sync / update 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "offer": Object {
        "ask_price": "0.76716532412734944381",
        "ask_price_sort": 0.7671653241273494,
        "bid_price": "1.3035",
        "bid_price_sort": 1.3035,
        "buyHowMuch": "100000000000000000",
        "buyHowMuch_filter": 100000000000000000,
        "buyWhichToken": "MKR",
        "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
        "id": 61209,
        "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
        "sellHowMuch": "130350000000000000",
        "sellHowMuch_filter": 130350000000000000,
        "sellWhichToken": "W-ETH",
        "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
        "status": undefined,
      },
      "offerType": "OFFERS/TYPE_BUY",
      "previousOfferState": undefined,
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/UPDATE_OFFER",
  },
]
`;

exports[`setOfferEpic new offer sync / set 1`] = `undefined`;

exports[`setOfferEpic new offer sync / set 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "offer": Object {
        "ask_price": "0.76716532412734944381",
        "ask_price_sort": 0.7671653241273494,
        "bid_price": "1.3035",
        "bid_price_sort": 1.3035,
        "buyHowMuch": "100000000000000000",
        "buyHowMuch_filter": 100000000000000000,
        "buyWhichToken": "MKR",
        "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
        "id": 61209,
        "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
        "sellHowMuch": "130350000000000000",
        "sellHowMuch_filter": 130350000000000000,
        "sellWhichToken": "W-ETH",
        "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
        "status": undefined,
      },
      "offerType": "OFFERS/TYPE_BUY",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/SET_OFFER",
  },
]
`;

exports[`setOfferEpic update sync / complete 1`] = `undefined`;

exports[`setOfferEpic update sync / complete 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "offerId": 61209,
      "offerType": "OFFERS/TYPE_BUY",
      "previousOfferState": undefined,
      "quoteToken": "W-ETH",
      "updatedOffer": Object {
        "ask_price": "0",
        "ask_price_sort": 0,
        "bid_price": "Infinity",
        "bid_price_sort": Infinity,
        "buyHowMuch": "0",
        "buyHowMuch_filter": 0,
        "buyWhichToken": "MKR",
        "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
        "id": 61209,
        "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
        "sellHowMuch": "130350000000000000",
        "sellHowMuch_filter": 130350000000000000,
        "sellWhichToken": "W-ETH",
        "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
        "status": undefined,
      },
    },
    "type": "OFFERS/OFFER_COMPLETELY_FILLED_IN",
  },
]
`;

exports[`setOfferEpic update sync / partial 1`] = `undefined`;

exports[`setOfferEpic update sync / partial 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "offer": Object {
        "ask_price": "0.76716532412734944381",
        "ask_price_sort": 0.7671653241273494,
        "bid_price": "1.3035",
        "bid_price_sort": 1.3035,
        "buyHowMuch": "100000000000000000",
        "buyHowMuch_filter": 100000000000000000,
        "buyWhichToken": "MKR",
        "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
        "id": 61209,
        "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
        "sellHowMuch": "130350000000000000",
        "sellHowMuch_filter": 130350000000000000,
        "sellWhichToken": "W-ETH",
        "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
        "status": undefined,
      },
      "offerType": "OFFERS/TYPE_BUY",
      "previousOfferState": undefined,
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/UPDATE_OFFER",
  },
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "offerId": 61209,
      "offerType": "OFFERS/TYPE_BUY",
      "previousOfferState": undefined,
      "quoteToken": "W-ETH",
      "updatedOffer": Object {
        "ask_price": "0.76716532412734944381",
        "ask_price_sort": 0.7671653241273494,
        "bid_price": "1.3035",
        "bid_price_sort": 1.3035,
        "buyHowMuch": "100000000000000000",
        "buyHowMuch_filter": 100000000000000000,
        "buyWhichToken": "MKR",
        "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
        "id": 61209,
        "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
        "sellHowMuch": "130350000000000000",
        "sellHowMuch_filter": 130350000000000000,
        "sellWhichToken": "W-ETH",
        "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
        "status": undefined,
      },
    },
    "type": "OFFERS/OFFER_PARTIALLY_FILLED_IN",
  },
]
`;

exports[`syncOffer initial sync 1`] = `
Object {
  "offer": Array [
    "130350000000000000",
    "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
    "100000000000000000",
    "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
    "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
    false,
    "0",
  ],
  "offerMeta": Object {
    "baseToken": "MKR",
    "offerType": "OFFERS/TYPE_BUY",
    "quoteToken": "W-ETH",
  },
}
`;

exports[`syncOffer initial sync 2`] = `Array []`;

exports[`syncOffer initial sync 3`] = `
Array [
  Array [
    "MKR",
    "W-ETH",
  ],
]
`;

exports[`syncOffer initial sync 4`] = `
Array [
  Array [
    Object {
      "buyHowMuch": "100000000000000000",
      "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
      "id": "61209",
      "offerType": "OFFERS/TYPE_BUY",
      "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
      "sellHowMuch": "130350000000000000",
      "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
      "syncType": "OFFERS/OFFER_SYNC_TYPE_INITIAL",
      "timestamp": false,
      "tradingPair": Object {
        "baseToken": "MKR",
        "quoteToken": "W-ETH",
      },
    },
  ],
]
`;

exports[`syncOffer new offer sync 1`] = `
Object {
  "offer": Array [
    "130350000000000000",
    "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
    "100000000000000000",
    "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
    "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
    false,
    "0",
  ],
  "offerMeta": Object {
    "baseToken": "MKR",
    "offerType": "OFFERS/TYPE_BUY",
    "quoteToken": "W-ETH",
  },
}
`;

exports[`syncOffer new offer sync 2`] = `Array []`;

exports[`syncOffer new offer sync 3`] = `
Array [
  Array [
    "MKR",
    "W-ETH",
  ],
]
`;

exports[`syncOffer new offer sync 4`] = `
Array [
  Array [
    Object {
      "buyHowMuch": "100000000000000000",
      "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
      "id": "61209",
      "offerType": "OFFERS/TYPE_BUY",
      "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
      "sellHowMuch": "130350000000000000",
      "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
      "syncType": "OFFERS/OFFER_SYNC_NEW_OFFER",
      "timestamp": false,
      "tradingPair": Object {
        "baseToken": "MKR",
        "quoteToken": "W-ETH",
      },
    },
  ],
]
`;

exports[`syncOffer update sync 1`] = `
Object {
  "offer": Array [
    "130350000000000000",
    "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
    "100000000000000000",
    "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
    "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
    false,
    "0",
  ],
  "offerMeta": Object {
    "baseToken": "MKR",
    "offerType": "OFFERS/TYPE_BUY",
    "quoteToken": "W-ETH",
  },
}
`;

exports[`syncOffer update sync 2`] = `Array []`;

exports[`syncOffer update sync 3`] = `
Array [
  Array [
    "MKR",
    "W-ETH",
  ],
]
`;

exports[`syncOffer update sync 4`] = `
Array [
  Array [
    Object {
      "buyHowMuch": "100000000000000000",
      "buyWhichTokenAddress": "0xaaf64bfcc32d0f15873a02163e7e500671a4ffcd",
      "id": "61209",
      "offerType": "OFFERS/TYPE_BUY",
      "owner": "0x2495eb6895c2e6c591ae9eb63a07b4a450623220",
      "previousOfferState": null,
      "sellHowMuch": "130350000000000000",
      "sellWhichTokenAddress": "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
      "syncType": "OFFERS/OFFER_SYNC_TYPE_UPDATE",
      "timestamp": false,
      "tradingPair": Object {
        "baseToken": "MKR",
        "quoteToken": "W-ETH",
      },
    },
  ],
]
`;

exports[`syncOffersEpic main 1`] = `
Object {
  "payload": Object {
    "baseToken": "MKR",
    "quoteToken": "W-ETH",
  },
  "type": Array [
    "OFFERS/SYNC_OFFERS_FULFILLED",
  ],
}
`;

exports[`syncOffersEpic main 2`] = `
Array [
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/SYNC_OFFERS_PENDING",
  },
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": "OFFERS/RESET_OFFERS",
  },
  Object {
    "payload": Object {
      "baseToken": "MKR",
      "quoteToken": "W-ETH",
    },
    "type": Array [
      "OFFERS/SYNC_OFFERS_FULFILLED",
    ],
  },
]
`;

exports[`syncOffersEpic main 3`] = `
Array [
  Array [
    10,
    "MKR",
    "W-ETH",
  ],
]
`;

exports[`syncOffersEpic main 4`] = `
Array [
  Array [
    10,
    "MKR",
    "W-ETH",
  ],
]
`;
